nbuild/03.the_templates.md

```diff
@@ -0,0 +1,75 @@
+
+A function is called for each of these steps and this template calls each of them in the above order.
+All these functions can be given as arguments, but they all have a default value (explained below). If a function is `None`, its step is skipped.
+
+### Fetch
+This step is used to download the source code. The default value is fetch() with no argument.
+
+### Extract
+This step is used to extract the downloaded source code. The default value is [flat_extract_all()](https://docs.raven-os.org/nbuild/master/stdlib.extract.html#stdlib.extract.flat_extract_all) with no argument.
+
+### Patch
+This step is used to patch the downloaded source code.
+The default value is [patch_all()](https://docs.raven-os.org/nbuild/master/stdlib.patch.html#stdlib.patch.patch_all) with no argument.
+
+* From this point on, the current working directory is changed in favor of build_folder (which defaults to `.`).
+If the directory pointed by build_folder doesn’t exist, it is created.
+This is useful for the few configure scripts that don’t work if they aren’t executed in a standalone directory.
+
+### Configure
+This step uses the `configure` script to configure the source code.
+The default value is [configure()](https://docs.raven-os.org/nbuild/master/stdlib.template.configure.html#stdlib.template.configure.configure) with no argument.
+
+When `build_folder` isn’t `.`, it is usually necessary to override this step with an other call to [configure()](https://docs.raven-os.org/nbuild/master/stdlib.template.configure.html#stdlib.template.configure.configure) with the argument `binary='../configure'`.
+
+### Compile
+This step compiles the source code.
+The default value is [make()](https://docs.raven-os.org/nbuild/master/stdlib.template.make.html#stdlib.template.make.make) with no argument.
+
+### Check
+This step runs the unit and integration tests.
+The default value is [make()](https://docs.raven-os.org/nbuild/master/stdlib.template.make.html#stdlib.template.make.make) with the arguments `'check'` and `fail_ok=True`.
+
+### Install
+This step installs the software in the install cache.
+The default value is [make()](https://docs.raven-os.org/nbuild/master/stdlib.template.make.html#stdlib.template.make.make) with the arguments `'install'` and `f'DESTDIR={stdlib.build.current_build().install_cache}'`.
+
+If this step is overwriten, the `DESTDIR` variable is crucial and should be reused.
+Otherwise, and unless the `configure` script has been configured to work without it, the installation can damage and overwrite parts of the host system.
+
+### Split
+This step is still a work in progress.
+The default value is `None`.
+
+### Dependency linking
+This step is still a work in progress.
+The default value is `None`.
+
+## 3.2 Examples of build manifests overwriting the templates
+### 3.2.1 Overwriting `autotools`
+The manifest of glibc is a good example showing how the steps `extract`, `install` and `split` from the `autotools` template can be overwriten.
```

Comment:
Pick an simpler manifest than glibc, like readline.

(Ndt: it would be great to inline the example and explain it, so that it is really clear what is happening and in which order)